firmware/Core/Src/telecommands/telecommand_args_helpers.c

```diff
@@ -83,6 +83,86 @@ uint8_t TCMD_extract_uint64_arg(const char *str, uint32_t str_len, uint8_t arg_i
     return 0;
 }
 
+/// @brief Extracts the longest substring of integer characters, starting from the beginning of the
+///     string, to a maximum length or until the first non-integer character is found.
+/// @param str Input string, starting with an integer
+/// @param str_len Max length of the input string
+/// @param result Pointer to the result
+/// @return 0 if successful, 1 if the string is empty, 2 if the string does not entirely encapsulate an integer
+uint8_t TCMD_ascii_to_int64(const char *str, uint32_t str_len, int64_t *result) {
+    if (str_len == 0) {
+        return 1;
+    }
+    uint64_t temp_result = 0;
+    uint32_t i = 0;
+    for (; i < str_len; i++) {
+        if (str[i] == '-' && i == 0) {
+            continue;
+        }
+        else if (str[i] < '0' || str[i] > '9') {
+            break;
+        }
+        temp_result = temp_result * 10 + (str[i] - '0');
+    }
+
+    if (str[0] == '-') {
+        temp_result *= -1;
+    }
+
+    if (i == 0) {
+        return 2;
+    }
+    else if ((str[i] < '0' || str[i] > '9') && str[i] != '\0') {
+        // check to make sure the string doesn't end with garbage characters
+        return 2;
+    }
+
+    *result = temp_result;
+    return 0;
+}
+
+/// @brief Extracts the nth comma-separated argument from the input string, assuming it's an int64
+/// @param str Input string
+/// @param str_len Length of the input string
+/// @param arg_index Index of the argument to extract (0-based)
+/// @param result Pointer to the result
+/// @return 0 if successful, 1 if the string is empty, 2 if the string does not contain enough arguments
+///        3 if the argument is not an integer, 4 for other error
+uint8_t TCMD_extract_int64_arg(const char *str, uint32_t str_len, uint8_t arg_index, int64_t *result) {
```

Comment:
Would love to see a unit test or two for this function!

Reply:
Done! Thank you!